lib/externalGraderCommon.js

```diff
@@ -117,11 +117,16 @@ module.exports.buildDirectory = function(dir, submission, variant, question, cou
                     // the directory hierarchy
                     const fullPath = path.join(dir, 'student', file.name);
                     const relPath = path.relative(path.join(dir, 'student'), fullPath);
-                    if (relPath.split(path.sep)[0] == '..' || path.isAbsolute(relPath)) {
+                    if (relPath.split(path.sep).includes('..') || path.isAbsolute(relPath)) {
```

Comment:
Is this change to catch potential security issues with `/../` in the middle of paths?

Reply:
Yes, keeping in the spirit of the original version that checked the first part only. Otherwise `foo/../../bar` for example would achieve the same effect as `../bar`. However since the name of the collected file is specified by the question, maybe this isn't so much a security issue as a sanity check on what the staff are doing?

Reply:
Because [`path.join()`](https://nodejs.org/api/path.html#path_path_join_paths) normalizes its output, both arguments to `path.relative()` will be normalized on the previous line. As I understand it, this means that `relPath` can only contain `..` segments at the start of the path, so I think this change has no effect.

I'm ok to make the change anyway, because it's harmless and maybe slightly clearer.

Reply:
Ah, that makes sense.